lib/srv/discovery/kube_integration_watcher.go

```diff
@@ -366,3 +379,25 @@ func (s *Server) getKubeIntegrationFetchers() []common.Fetcher {
 func (s *Server) getKubeNonIntegrationFetchers() []common.Fetcher {
 	return s.getKubeFetchers(false)
 }
+
+func versionGetterForProxy(ctx context.Context, proxyPublicAddr string) (version.Getter, error) {
+	proxyClt, err := webclient.NewReusableClient(&webclient.Config{
+		Context:   ctx,
+		ProxyAddr: proxyPublicAddr,
+	})
+	if err != nil {
+		return nil, trace.Wrap(err, "failed to build proxy client")
+	}
+
+	baseURL, err := url.Parse(fmt.Sprintf("https://%s/webapi/automaticupgrades/channel/%s", proxyPublicAddr, automaticupgrades.DefaultChannelName))
```

Comment:
instead of doing this, build the baseURL using:


```suggestion
	baseURL := &url.URL{
		Scheme:  "https",
		Host:    proxyPublicAddr,
		RawPath: path.Join("/webapi/automaticupgrades/channel", automaticupgrades.DefaultChannelName),
	}
```